tests/test_appmesh.py

```diff
@@ -16,21 +20,96 @@
+              'c7n:MatchedFilters': ['tag:MyTagName', 'spec.egressFilter.type'],
+              'createdAt': '2023-11-03T02:36:27.877000+00:00',
+              'lastUpdatedAt': '2023-11-03T02:36:27.877000+00:00',
+              'meshName': 'm2',
+              'meshOwner': '123456789012',
+              'metadata': {'arn': 'arn:aws:appmesh:eu-west-2:123456789012:mesh/m2',
+                           'createdAt': '2023-11-03T02:36:27.877000+00:00',
+                           'lastUpdatedAt': '2023-11-03T02:36:27.877000+00:00',
+                           'meshOwner': '123456789012',
+                           'resourceOwner': '123456789012',
+                           'uid': '1013368a-8d58-4b30-b08e-67347af88525',
+                           'version': 1},
+              'resourceOwner': '123456789012',
+              'spec': {'egressFilter': {'type': 'DROP_ALL'},
+                       'serviceDiscovery': {'ipPreference': 'IPv6_PREFERRED'}},
+              'status': {'status': 'ACTIVE'},
+              'version': 1},
+             ],
+            resources,
+        )
 
         # These assertions are necessary to be sure that the "get_arns" function is correctly
         # deriving the ARN.
         # See the documentation on the "arn" field in appmesh.py.
         arns = p.resource_manager.get_arns(resources)
-        self.assertIn('arn:aws:appmesh:eu-west-2:123456789012:mesh/m1', arns)
-        self.assertIn('arn:aws:appmesh:eu-west-2:123456789012:mesh/m2', arns)
+        self.assertEqual(
+            [
+                'arn:aws:appmesh:eu-west-2:123456789012:mesh/m2',
+            ],
+            arns,
+        )
+
+        # The "placebo" testing library doesn't allow us to make assertions
+        # linking specific api's calls to the specific mock response file
+        # that will serve that request. So we will compensate here by
+        # making an assertion about all the api calls and the order
+        # of calls that must be made.
+        self.assertEqual(
+            [
+                {'operation': 'ListMeshes', 'params': {}, 'service': 'appmesh'},
+                {'operation': 'DescribeMesh', 'params': {'meshName': 'm1'}, 'service': 'appmesh'},
+                {'operation': 'DescribeMesh', 'params': {'meshName': 'm2'}, 'service': 'appmesh'},
+                {'operation': 'DescribeMesh', 'params': {'meshName': 'm3'}, 'service': 'appmesh'},
+                {
+                    'operation': 'GetResources',
+                    'params': {
+                        'ResourceARNList': [
+                            'arn:aws:appmesh:eu-west-2:123456789012:mesh/m1',
+                            'arn:aws:appmesh:eu-west-2:123456789012:mesh/m2',
+                            'arn:aws:appmesh:eu-west-2:123456789012:mesh/m3',
+                        ]
+                    },
+                    'service': 'resourcegroupstaggingapi',
+                },
+            ],
+            captor.calls,
+        )

```

Comment:
I like this level of detail from an observer standpoint, but I'm not sure what it functionally buys us to assert this during every test run. if we've already asserted that the shape/content of the policy results match expectations, do we need to also tick off each botocore call that got us there?

(This is a question not just for you, but for other maintainers. I can honestly see it from both sides, it's just a tradeoff.)

Reply:
It found bugs in my code eve. Though I had other tests.